### PR TITLE
Various quality fixes

### PR DESCRIPTION
**Pull request by meyerbaptiste**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

File: src/Bridge/Doctrine/MongoDbOdm/Filter/BooleanFilter.php [hunk 1]
```diff
@@ -36,7 +36,7 @@ final class BooleanFilter extends AbstractFilter
 {
     use BooleanFilterTrait;
 
-    const DOCTRINE_BOOLEAN_TYPES = [
+    public const DOCTRINE_BOOLEAN_TYPES = [
```
Comment (soyuka) [inline src/Bridge/Doctrine/MongoDbOdm/Filter/BooleanFilter.php]: I'm -1 on this `public const` thing it's just noise IMO. Symfony is not using it for now?

File: src/Metadata/Property/PropertyMetadata.php [hunk 2]
```diff
@@ -281,14 +255,32 @@ public function withAttributes(array $attributes): self
...
-     * @return string|null
+     * @deprecated since version 2.4, to be removed in 3.0.
      */
-    public function isChildInherited()
+    public function isChildInherited(): ?string
     {
-        return $this->childInherited;
+        @trigger_error(sprintf('The use of "%1$s::isChildInherited()" is deprecated since 2.4 and will be removed in 3.0. Use "%1$s::getChildInherited()" or "%1$s::hasChildInherited()" directly instead.', __CLASS__), E_USER_DEPRECATED);
+
+        return $this->getChildInherited();
```
Comment (soyuka) [inline src/Metadata/Property/PropertyMetadata.php]: why the change?

Comment (meyerbaptiste) [reply src/Metadata/Property/PropertyMetadata.php]: The name of the method and the return type are inconsistent... It's a nonsense.

File: src/Bridge/Doctrine/MongoDbOdm/Filter/BooleanFilter.php (hunk 1 above)
Comment (meyerbaptiste) [reply src/Bridge/Doctrine/MongoDbOdm/Filter/BooleanFilter.php]: We should now pay attention to the visibility of our constants and define them explicitly. It's by default in the PHP-CS-Fixer ruleset `@PHP71Migration`.

Comment (teohhanhui) [reply src/Bridge/Doctrine/MongoDbOdm/Filter/BooleanFilter.php]: > It's by default in the PHP-CS-Fixer ruleset @PHP71Migration.

But we are already using both `@PHP71Migration` and `@PHP71Migration:risky` rulesets?

Comment (teohhanhui) [reply src/Bridge/Doctrine/MongoDbOdm/Filter/BooleanFilter.php]:
```
visibility_required [@PSR2, @Symfony, @PhpCsFixer, @PHP71Migration]

Visibility MUST be declared on all properties and methods; abstract and final MUST be declared before the visibility; static MUST be declared after the visibility.

Configuration options:

- elements (a subset of ['property', 'method', 'const']): the structural elements to fix (PHP >= 7.1 required for const); defaults to ['property', 'method']
```

So it's not enabled by default. If we're going to make this change, we must change that option.

Comment (soyuka) [reply src/Bridge/Doctrine/MongoDbOdm/Filter/BooleanFilter.php]: By default constants are public and it won't change, we should defined them if they're not public IMO.

https://secure.php.net/manual/en/migration71.new-features.php#migration71.new-features.class-constant-visibility

Comment (meyerbaptiste) [reply src/Bridge/Doctrine/MongoDbOdm/Filter/BooleanFilter.php]: Yep but because of the alphabetical order in the configuration, the `@Symfony` ruleset has the advantage and it uses the default configuration: `method` + `property` vs. `method` + `property` + `const` for the `@ PHP71Migration` ruleset.

Comment (meyerbaptiste) [reply src/Bridge/Doctrine/MongoDbOdm/Filter/BooleanFilter.php]: > By default constants are public and it won't change

By default method are public but you define a visibility explicitly even if it is public.

Comment (teohhanhui) [reply src/Bridge/Doctrine/MongoDbOdm/Filter/BooleanFilter.php]: Okay, let's add the correct configuration for this rule then.

Comment (teohhanhui) [reply src/Bridge/Doctrine/MongoDbOdm/Filter/BooleanFilter.php]: > Yep but because of the alphabetical order in our configuration, the @Symfony ruleset has the advantage

That is indeed unexpected, but might as well since I don't think we would ever find a "correct" order. :laughing:

**Comment by teohhanhui** [comment]
Anyway, I suspect this PR should target `2.4` first, to reduce future merge conflicts.

**Comment by meyerbaptiste** [comment]
Base branch changed.

File: src/Bridge/Doctrine/MongoDbOdm/Filter/BooleanFilter.php (hunk 1 above)
Comment (meyerbaptiste) [reply src/Bridge/Doctrine/MongoDbOdm/Filter/BooleanFilter.php]: I just updated the `.php_cs.dist` file.

**Comment by dunglas** [comment]
Thanks @meyerbaptiste
